RELEASE.md

```diff
@@ -0,0 +1,100 @@
+# sBTC: Release Process
+
+## Executive summary
+
+This release process targets the following goals:
+
+1. Ensure there exists a provable chain of trust connecting the source code to
+   the corresponding artifacts.
+1. Ensure there exists a clear _separation of duties_ between those who _write_
+   the code and those who _release_ it (and announce its release).
+
+It implements (1) by leveraging GitHub rulesets for branch/tag protection and
+attestation for artifacts and (2) through the announcement process (described
+below).
+
+## [sBTC Core developers] Creating a new release
+
+An sBTC release is a tagged commit on the `main` branch.
+
+Any commit to `main` SHALL require at least one review through a GitHub pull
+request. Before merging the pull request, all tests SHALL pass.
+
+Tags SHALL be named according to [semantic versioning][0].
+
+[GitHub rulesets][1] ensure that only a subset of sBTC core developers can
+create a `git tag`. Creating a tag SHOULD require 4-eyes (as of February 2025,
+this is not yet possible).
+
+Once a tag is created, a [GitHub deployment environment][2] will build and
+publish any corresponding artifact. The deployment environment SHALL require a
+review from a subset of sBTC core developers before executing. The use of
+deployment environment ensures that any credential required to publish artifacts
+is gated behind the review process (e.g., Docker Hub credentials, until [OIDC
+identities will be supported][4]).
+
+All artifacts SHALL be [attested][3] so that their build provenance can be
+established. This way, downstream users (e.g., sBTC signers) will be able to
+cryptographically verify that an artifact (e.g., a Docker release) has been
+built and published through GitHub actions.
+
+All artifacts SHALL be addressed through their cryptographic digest (e.g., `git
+commit` or Docker image digest), in addition to their label (e.g., the `git
+tag`).
+
+To improve quality of life, the release notes SHALL include breaking changes (if
+any), upgrade migrations (if any), and a link to the relevant artefacts (e.g.,
+Docker images).
+
+## [sBTC Comms] Announcing a new release
+
+After a new release has been created, the sBTC Comms team will inform the sBTC
+signers and provide the appropriate update instructions.
+
+The sBTC Comms team SHALL NOT intersect with the sBTC development team. This
+ensures clear separation of duties and, for instance, prevents a rogue core
+developer from "convincing" the sBTC signers from deploying a tampered release.
+
+At all times, there shall be at least two members of the sBTC Comms team in any
+channel including an sBTC signer (similarly to the 4-eyes process for releases).
```

Comment:
These two paragraphs feel like they belong under a more general "Communication Guidelines" section or something, as they're not related only to release announcements but also tech support etc.

Reply:
I think you have a point, but let me keep it here for simplicity for now and we can de-structure it later on once it's clear how we wanna run comms.